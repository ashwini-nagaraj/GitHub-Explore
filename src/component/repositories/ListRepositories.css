.repositories {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  margin-top: 20px;
  align-items: center;
  justify-content: center;
}
i {
  color: rgb(71, 67, 67);
}
.repo {
  width: 80%;
  border: solid 1px #ccc;
  background-color: rgb(235, 235, 235);
  border-radius: 5px;
  padding: 10px;
  margin-bottom: 10px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  transition: transform 0.5s ease-in;
}

.title {
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 1;
  line-height: 1.5rem;
  max-height: 1.5rem;
  font-weight: bold;
  cursor: pointer;
}
.title > span {
  font-weight: bolder;
  cursor: pointer;
}

.error {
  font-size: 14px;
  font-weight: bold;
}

.link {
  text-decoration: none;
}

.description {
  font-size: 14px;
  color: black;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 1;
  line-height: 1.5rem;
  max-height: 1.5rem;
}
.subitem {
  display: flex;
  align-items: center;
  font-size: 12px;
  margin-top: 10px;
  margin-right: 10px;
}
