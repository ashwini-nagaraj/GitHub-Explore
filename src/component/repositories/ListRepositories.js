import React from "react";
import Repository from "./Repository";
import "./ListRepositories.css";

const Repositories = ({ repos =[] , user =[]}) => {
  return (
    <section data-testid="section" className="repositories">
      {repos.length > 0 ? (
        <>
          {repos.map((repo) => (
            <Repository  data-testid="repo-test"  key={repo.id} repo={repo} user={user} />
          ))}
        </>
      ) : (
        <span data-testid="error" className="error">
          You sure what you are looking for? Looks like there is no repository
          with this name
        </span>
      )}
    </section>
  );
};

export default Repositories;
