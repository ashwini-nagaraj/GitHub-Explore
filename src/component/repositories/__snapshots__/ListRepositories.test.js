import React from "react";
import ReactDOM from "react-dom";
import ListRepositories from "../ListRepositories";
import { render , cleanup} from "@testing-library/react";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("renders without crashig", () => {
  const div = document.createElement("div");
  ReactDOM.render(<ListRepositories></ListRepositories>, div);
});

it("renders loading correctly", () => {
  const { getByTestId } = render(
    <ListRepositories>
    </ListRepositories>
  );
});
