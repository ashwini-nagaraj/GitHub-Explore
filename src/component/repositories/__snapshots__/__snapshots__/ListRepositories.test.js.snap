// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<section
  className="repositories"
  data-testid="section"
>
  <span
    className="error"
    data-testid="error"
  >
    You sure what you are looking for? Looks like there is no repository with this name
  </span>
</section>
`;
